staging/src/k8s.io/apiserver/pkg/cel/library/authz.go

```diff
@@ -0,0 +1,573 @@
+//
+// Examples:
+//
+//	authorizer.group('apps').resource('deployments').namespace('test') // returns a ResourceCheck for 'deployments' in the 'test' namespace
+//	authorizer.group('').resource('pods').namespace('default') // returns a ResourceCheck for 'pods' in the 'default' namespace
+//
+// name
+//
+// Returns a ResourceCheck configured to check authorization for a particular resource name.
+//
+//	<ResourceCheck>.path(<name>) <ResourceCheck>
+//
+// Examples:
+//
+//	authorizer.group('apps').resource('deployments').namespace('test').name('backend') // returns a ResourceCheck for the 'backend' 'deployments' resource in the 'test' namespace
+//
+// check
+//
+// For PathCheck, checks if the principal (user or service account) that sent the request is authorized for the HTTP request verb of the path.
+// For ResourceCheck, checks if the principal (user or service account) that sent the request is authorized for the API verb and the configured authorization checks of the ResourceCheck.
+//
+//	<PathCheck>.check(<check>) <Decision>
+//	<ResourceCheck>.check(<check>) <Decision>
+//
+// Examples:
+//
+//	authorizer.group('').resource('pods').namespace('default').check('create') // Checks if the principal (user or service account) is authorized create pods in the 'default' namespace.
+//	authorizer.path('/healthz').check('get') // Checks if the principal (user or service account) is authorized to make HTTP GET requests to the /healthz API path.
+//
+// allowed
+//
+// Returns true if the authorizer's decision for the check is "allow".  Note that if the authorizer's decision is
+// "no opinion", that both the 'allowed' and 'denied' functions will return false.
+//
+//	<Decision>.allowed() <bool>
+//
+// Examples:
+//
+//	authorizer.group('').resource('pods').namespace('default').check('create').allowed() // Returns true if the principal (user or service account) is allowed create pods in the 'default' namespace.
+//	authorizer.path('/healthz').check('get').allowed()  // Returns true if the principal (user or service account) is allowed to make HTTP GET requests to the /healthz API path.
+//
+// denied
+//
+// Returns true if the authorizer's decision for the check is "deny".  Note that if the authorizer's decision is
+// "no opinion", that both the 'allowed' and 'denied' functions will return false.
+//
+//	<Decision>.denied() <bool>
+//
+// Examples:
+//
+//	authorizer.group('').resource('pods').namespace('default').check('create').denied() // Returns true if the principal (user or service account) is denied authorization to create pods in the 'default' namespace.
+//	authorizer.path('/healthz').check('get').denied()  // Returns true if the principal (user or service account) is denied authorization to make HTTP GET requests to the /healthz API path.
+//
+// reason
+//
+// Returns a string reason for the authorization decision
+//
+//	<Decision>.reason() <string>
+//
+// Examples:
+//
+//	authorizer.path('/healthz').check('GET').reason()
```

Comment:
cc @DangerOnTheRanger

Yes, we'll omit `authorizer` from `messageExpression`

Reply:
The KEP does mention the possibility of caching decisions for identical checks within a given policy evaluation. That may still be worth considering, especially if multiple expressions within a policy reference the same decision.

Reply:
This sounds like something worth doing. Maybe as a follow up PR? It needs to wired in carefully to be shared across expressions. It would be great to get it in before (or with) the PR for `messageExpression`, which might not be till 1.28.